core/arch/arm/plat-ls/main.c

```diff
@@ -94,3 +167,36 @@ void console_flush(void)
 {
 	ns16550_flush(console_base());
 }
+
+void main_init_gic(void)
+{
+	vaddr_t gicc_base;
+	vaddr_t gicd_base;
+
+	gicc_base = (vaddr_t)phys_to_virt(GIC_BASE + GICC_OFFSET,
+					  MEM_AREA_IO_SEC);
+	gicd_base = (vaddr_t)phys_to_virt(GIC_BASE + GICD_OFFSET,
+					  MEM_AREA_IO_SEC);
+
+	if (!gicc_base || !gicd_base)
+		panic();
+
+	/* Initialize GIC */
+	gic_init(&gic_data, gicc_base, gicd_base);
+
+	itr_init(&gic_data.chip);
+}
+
+static void platform_tee_entry_fast(struct thread_smc_args *args)
```

Comment:
`OPTEE_SMC_BOOT_SECONDARY` is a generic OPTEE SMC.
Could you change `tee_entry_fast()` to handle `OPTEE_SMC_BOOT_SECONDARY` and call `generic_smp_boot()` when `CFG_BOOT_SECONDARY_REQUEST` is enable ?
This will factorize platform code (`plat-imx/main.c` to update).